core/server/master/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1242,14 +1241,14 @@ public int getNumberOfPinnedFiles() {
    * permission on the parent of the path.
    *
    * @param path the path to delete
-   * @param recursive if true, will delete all its children
+   * @param options method options
    * @throws DirectoryNotEmptyException if recursive is false and the file is a nonempty directory
    * @throws FileDoesNotExistException if the file does not exist
    * @throws IOException if an I/O error occurs
    * @throws AccessControlException if permission checking fails
    * @throws InvalidPathException if the path is invalid
    */
-  public void delete(AlluxioURI path, boolean recursive)
+  public void delete(AlluxioURI path, alluxio.master.file.options.DeleteOptions options)
```

Comment:
NIT: since you have removed `import alluxio.underfs.options.DeleteOptions` and inline it in the code, it is fine to `import alluxio.master.file.options.DeleteOptions` and simplify the signature here a little bit?